app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -15,9 +15,48 @@ class ManageIQ::Providers::Redhat::InfraManager < ManageIQ::Providers::InfraMana
   include_concern :ApiIntegration
   include_concern :VmImport
 
+  has_many :cloud_tenants, :foreign_key => :ems_id, :dependent => :destroy
+
+  include HasNetworkManagerMixin
+
   supports :provisioning
   supports :refresh_new_target
 
+  before_update :ensure_managers
+  before_update :ensure_managers_zone_and_provider_region
+
+  def ensure_network_manager
+    providers = ovirt_services.collect_external_network_providers
+
+    unless providers.blank?
+      providers = providers.sort_by(&:name)
```

Comment:
Can we consider also other properties of the provider such as: status or last access time to select the better match ?

Reply:
Currently, oVirt api doesn't provide us any of these proposed properties.